### ISSUE TITLE
[BUG] Building against static LibRaw no longer finds static jasper dependency

### ISSUE THREAD

**Issue by nrusch** [OpenImageIO/oiio#3209]
**Describe the bug**

It looks like the OIIO build stopped being able to properly link against a static LibRaw and all of its dependencies between 2.2.15.0 and 2.2.16.0.

I'm building against a static build of LibRaw 0.20.2 (which was built with `--enable-jpeg --enable-jasper --enable-lcms`), and in OIIO 2.2.15.0, the static `jasper` dependency is resolved and linked appropriately:

```
-- Found LibRaw: /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw.a (found version "0.20.2") 
-- Found JPEG: /luma/soft/rez/packages/jpeg/9.c/platform-linux/gcc-6.3.1/lib/libjpeg.a (found version "90") 
-- Found Jasper: /luma/soft/rez/packages/jasper/2.0.16/platform-linux/gcc-6.3.1/lib64/libjasper.a (found version "2.0.16") 
-- Found LibRaw 0.20.2 
--     LibRaw_INCLUDE_DIR = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/include
--     LibRaw_LIBRARIES = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw.a
--     LibRaw_r_LIBRARIES = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw_r.a;/luma/soft/rez/packages/jasper/2.0.16/platform-linux/gcc-6.3.1/lib64/libjasper.a;/luma/soft/rez/packages/jpeg/9.c/platform-linux/gcc-6.3.1/lib/libjpeg.a;/luma/soft/rez/packages/lcms/2.9/platform-linux/gcc-6.3.1/lib/liblcms2.a
```

However, building 2.2.16.0+ with the same dependencies/build configuration no longer resolves all of LibRaw's second-order static dependencies:

```
-- Found LibRaw: /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw.a (found version "0.20.2") 
-- Found LibRaw 0.20.2 
--     LibRaw_INCLUDE_DIR = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/include
--     LibRaw_LIBRARIES = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw.a
--     LibRaw_r_LIBRARIES = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw_r.a;/luma/soft/rez/packages/lcms/2.9/platform-linux/gcc-6.3.1/lib/liblcms2.a
```

This results in link errors for anything trying to use `libOpenImageIO`, e.g:
```
[ 83%] Linking CXX executable ../../bin/idiff
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_fdopen'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_image_destroy'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_close'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_image_readcmpt'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_image_decode'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_fopen'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_memopen'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_matrix_create'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_init'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_matrix_destroy'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_seek'
collect2: error: ld returned 1 exit status
```

From looking through the commit history, I'm fairly certain the culprit is in [8ede4690927e3074dab379379ac90e9547fdfb71](https://github.com/OpenImageIO/oiio/commit/8ede4690927e3074dab379379ac90e9547fdfb71#diff-c470226216e28a3919497ddd3194dcda83426b8edb884e62648be47d1b99d36e).

It's somewhat unclear why the search for Jasper was removed... are you able to provide any more context here?

Thanks.

**Comment by lgritz** [comment]
> It's somewhat unclear why the search for Jasper was removed... are you able to provide any more context here?

I don't remember, specifically. But perhaps I can make a good guess:

A long, long time ago, OIIO used Jasper to read JPEG-2000 files (beginning in 2009). At some 2011, we switched to OpenJpeg.

MAYBE... I saw these lines, thought "We don't use Jasper anymore, we can remove this search for it, it's left over from an earlier era," and because I hadn't yet had my morning coffee, it did not occur to me that it was still necessary because of a transitive dependency through LibRaw? And because I don't tend to build static libraries, I had not noticed the breakage?

I'll restore it. Sorry about that.
